qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py

```diff
@@ -75,8 +77,12 @@ def __init__(
         """
         super().__init__(neural_network, loss, optimizer, warm_start, initial_point)
         self._one_hot = one_hot
+        # encodes the target data if categorical
+        self._target_encoder = OneHotEncoder(sparse=False) if one_hot else LabelEncoder()
 
     def fit(self, X: np.ndarray, y: np.ndarray):  # pylint: disable=invalid-name
+        y = self._fit_and_encode_categorical(y)
```

Comment:
I'd rename to `_fit_and_encode_labels` or `_fit_transform_lables` with a stress on labels.